### PR TITLE
Package io.github.davejarvis.kmcaster

### PR DESCRIPTION
**Pull request by Yajo**
Java-based on-screen display (OSD) for keyboard and mouse events.

This program displays keyboard and mouse events for the purpose of screencasting.

According to https://github.com/DaveJarvis/kmcaster/issues/1#issuecomment-670951972, upstream maintainer is not interested in maintaining flatpak, but neither opposed to it. I took the initiative.

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build Icon=io.github.davejarvis.kmcaster

**Comment by Yajo** [comment]
I don't understand those logs. Do they mean the XML file must be gzipped? 

Also I'd prefer to build the jar file from sources, rather than just downloading it. Any ideas on how to do it?

Thanks!

File: io.github.davejarvis.kmcaster.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+id: io.github.davejarvis.kmcaster
+runtime: org.freedesktop.Platform
+runtime-version: "20.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk
+command: /app/bin/run.sh
+finish-args:
+  - --env=PATH=/app/jre/bin:/usr/bin
+  - --socket=x11
```
Comment (nedrichards) [inline io.github.davejarvis.kmcaster.yaml:10]: you should also add --share=ipc if you're using x11 to get high performance graphics

**Comment by nedrichards** [comment]
that error means that there's a problem in the final process that takes the icons, desktop file and appdata file and compiles them together to build the data that's used in software stores. Basically, make sure everything is being exported into the correct locations with the correct names.

You absolutely should be able to compile and deploy some java inside this system. It might be a bit annoying as there aren't any helper scripts for java in https://github.com/flatpak/flatpak-builder-tools but we would absolutely recommend giving it a go.

**Comment by Yajo** [comment]
bot, build io.github.davejarvis.kmcaster

**Comment by Yajo** [comment]
Other problems should be solved. About compiling in the flatpak... As a self-note:

I added this to the manifest:

```yaml

  - name: kmcaster-sources
    buildsystem: simple
    build-commands:
      - env JAVA_HOME=/usr/lib/sdk/openjdk/jvm/openjdk-15 /usr/lib/sdk/openjdk/bin/gradle jar
    sources:
      - type: archive
        url: https://github.com/DaveJarvis/kmcaster/archive/refs/tags/1.0.3.tar.gz
        sha256: 9795c1953554fd50e840056647be4119b906f622a80dd67990b30e3914062030
```

(I found out [here the SDK and gradle placement](https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk/blob/652cfb9fd62ee1f80ffc8998a7d95848627d6e5b/org.freedesktop.Sdk.Extension.openjdk.yaml#L128-L140))

It fails with:

```
Emptying app dir 'build-dir'
Downloading sources
Starting build of io.github.davejarvis.kmcaster
Cache hit for openjdk, skipping build
Cache miss, checking out last cache hit
========================================================================
Building module kmcaster-sources in /var/home/yajo/mydevel/flathub/.flatpak-builder/build/kmcaster-sources-12
========================================================================
Running: env JAVA_HOME=/usr/lib/sdk/openjdk/jvm/openjdk-15 /usr/lib/sdk/openjdk/bin/gradle jar

Welcome to Gradle 6.3!

Here are the highlights of this release:
 - Java 14 support
 - Improved error messages for unexpected failures

For more details see https://docs.gradle.org/6.3/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Build file '/run/build/kmcaster-sources/build.gradle' line: 48

* What went wrong:
Could not determine the dependencies of task ':jar'.
> Could not resolve all files for configuration ':runtimeClasspath'.
   > Could not resolve info.picocli:picocli:4.4.0.
     Required by:
         project :
      > Could not resolve info.picocli:picocli:4.4.0.
         > Could not get resource 'https://repo.maven.apache.org/maven2/info/picocli/picocli/4.4.0/picocli-4.4.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/info/picocli/picocli/4.4.0/picocli-4.4.0.pom'.
               > repo.maven.apache.org: Temporary failure in name resolution
      > Could not resolve info.picocli:picocli:4.4.0.
         > Could not get resource 'https://jcenter.bintray.com/info/picocli/picocli/4.4.0/picocli-4.4.0.pom'.
            > Could not GET 'https://jcenter.bintray.com/info/picocli/picocli/4.4.0/picocli-4.4.0.pom'.
               > jcenter.bintray.com: Temporary failure in name resolution
   > Could not resolve com.1stleg:jnativehook:2.1.0.
     Required by:
         project :
      > Could not resolve com.1stleg:jnativehook:2.1.0.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/1stleg/jnativehook/2.1.0/jnativehook-2.1.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/1stleg/jnativehook/2.1.0/jnativehook-2.1.0.pom'.
               > repo.maven.apache.org
      > Could not resolve com.1stleg:jnativehook:2.1.0.
         > Could not get resource 'https://jcenter.bintray.com/com/1stleg/jnativehook/2.1.0/jnativehook-2.1.0.pom'.
            > Could not GET 'https://jcenter.bintray.com/com/1stleg/jnativehook/2.1.0/jnativehook-2.1.0.pom'.
               > jcenter.bintray.com

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
Error: module kmcaster-sources: El proceso hijo terminó con el código 1
```

It seems like the builder process cannot access the Internet, and that's why stuff like https://github.com/flatpak/flatpak-builder-tools is needed.

I have no knowledge of Gradle, but it seems it is not supported currently: https://github.com/flatpak/flatpak-builder-tools/issues/37

Thus, for the time being, the simplest way to package this is what I'm doing here: download the precompiled jar.

It seems it only works on aarch64, though. If anybody would want to fix https://github.com/flatpak/flatpak-builder-tools/issues/37 for me, I'd be able to use it. Otherwise please could you help me to know how to limit available arches to keep advancing?

**Comment by Yajo** [comment]
bot, build io.github.davejarvis.kmcaster

**Comment by Yajo** [comment]
bot, build io.github.davejarvis.kmcaster

**Comment by Yajo** [comment]
bot, build io.github.davejarvis.kmcaster

**Comment by Yajo** [comment]
It works!

**Pull request closed without merging**